prov/util/src/util_wait.c

```diff
@@ -185,6 +187,7 @@ int ofi_wait_fd_add(struct util_wait *wait, int fd, ofi_wait_fd_try_func try,
 	int ret = 0;
 
 	fastlock_acquire(&wait_fd->lock);
+	ofi_atomic_inc32(&wait_fd->ref);
 	if (dlist_find_first_match(&wait_fd->fd_list, ofi_wait_fd_match, &fd)) {
 		FI_DBG(wait->prov, FI_LOG_EP_CTRL,
 			"wait_fd already added to util_wait fd_list \n");
```

Comment:
Here, if we find a match, we need to increment the fd_entry ref count, otherwise new entries should set the fd_entry ref = 1.

Reply:
Done